protocols/pop3/src/main/java/org/apache/james/protocols/pop3/core/MessageMetaDataUtils.java

```diff
@@ -28,14 +29,14 @@
 public class MessageMetaDataUtils {
 
     /**
-     * Returns the {@link MessageMetaData} for the given message number or <code>null</code> if it can not be 
-     * found.
+     * Returns the {@link MessageMetaData} for the given message number.
+     * @throws IOException if it can not be found.
      */
-    public static MessageMetaData getMetaData(POP3Session session, int number) {
+    public static MessageMetaData getMetaData(POP3Session session, int number) throws IOException {
```

Comment:
I would promote this to the `POP3Session` rather than adding an artificial "utils" class

`POP3Session::getMetadation(int number)`

Reply:
This utility class was already there, but it should not be hard to move its content to where it is actually needed.